### PR TITLE
Add GitHub Actions release workflow

### LINKED ISSUE #127: Update release steps
Update release steps, similar to https://github.com/E3SM-Project/e3sm_diags/pull/420.

### LINKED ISSUE #131: Add release workflow for publishing docs and Anaconda package
Similar to https://github.com/E3SM-Project/e3sm_diags/blob/master/.github/workflows/release_workflow.yml

### PR DESCRIPTION
**Pull request by tomvothecoder**
This PR adds the `release_workflow.yml` file, which publishes docs and the Anaconda package when a new release is tagged.

- Closes #131 
- Closes #127 

### Checklist
- [x] Add `release_workflow.yml` for GitHub Actions
- [x] Update docs, project standards page for CI/CD
- [x] Update docs for publishing Anaconda packages
- [x] Update docs for versioning docs

### Examples from my fork for `v1.0.2`:
- passing workflow: https://github.com/tomvothecoder/zstash/actions/runs/709431456
- docs: https://tomvothecoder.github.io/zstash/_build/html/v1.0.2/index.html
- Anaconda publishing: https://anaconda.org/tomvothecoder/zstash
   - It still says `v1.0.1` because I didn't change the version in `meta.yml`

### CODE REVIEW

File: docs/source/dev_guide/release.rst [hunk 1]
```diff
@@ -10,50 +10,70 @@ In this guide, we'll cover:
 * Creating a New Version of the Documentation
```
Comment (tomvothecoder) [inline docs/source/dev_guide/release.rst:10]: Ported from `e3sm-diags` with some modifications for `zstash` (e.g. no Docker)

File: .github/workflows/release_workflow.yml [hunk 2]
```diff
@@ -0,0 +1,117 @@
+name: CI/CD Release Workflow
```
Comment (tomvothecoder) [inline .github/workflows/release_workflow.yml:1]: Ported content from `e3sm-diags`

**Comment by tomvothecoder** [comment]
@forsyth2 This PR is ready for review. I've provided links to all the relevant release workflow outputs.

File: docs/source/dev_guide/project-standards.rst [hunk 3]
```diff
@@ -152,12 +152,26 @@ Run a tool
 Continuous Integration / Continuous Delivery (CI/CD)
 ----------------------------------------------------
 
-This project uses `GitHub Actions <https://github.com/E3SM-Project/zstash/actions>`_ to run the CI/CD build. The build is triggered by Git ``pull_request`` and ``push`` (merging PRs) events to the the main repo's ``master``.
+This project uses `GitHub Actions <https://github.com/E3SM-Project/zstash/actions>`_ to run two CI/CD workflows.
 
-The CI/CD workflow has two jobs:
+1. CI/CD Workflow
```
Comment (forsyth2) [inline docs/source/dev_guide/project-standards.rst:157]: Is there another name we can come up with so the two workflows aren't "workflow" and "release workflow"? (e.g., this one should have an adjective too).

File: docs/source/dev_guide/release.rst [hunk 4]
```diff
@@ -10,50 +10,70 @@ In this guide, we'll cover:
...
+    ::
 
-5. Commit and push your changes.
+        __version__ = 'v1.0.1'
+
+4. Change the ``version`` and ``git_rev`` tag in ``conda/meta.yaml``.
+``version`` is what the version of the software will be on Anaconda and
+``git_rev`` is the tag that we'll setup on GitHub in the next section.
+
+When running ``conda build``, ``conda`` will download the code tagged by ``git_rev``.
```
Comment (forsyth2) [inline docs/source/dev_guide/release.rst:53]: This seems like it would be better in a note box?

**Review by forsyth2 — APPROVED** [review]
Looks good to me, thanks. Just a couple minor comments.

**Pull request merged**